### PR TITLE
feat: add C implementation for `math/base/special/asec`

### LINKED ISSUE #861: [RFC]: Add C implementation for `@stdlib/math/base/special/asec`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/asec`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/asec).

```c
double stdlib_base_asec( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #861 .

## Description

> What is the purpose of this pull request?

This pull request:Add C implementation for @stdlib/math/base/special/asec

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #861 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/asec/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2022 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/README.md]:
```suggestion
Copyright (c) 2022 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asec/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asec/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+// MAIN //
+
+bench(pkg + '::native', opts, function benchmark(b) {
+	var y;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = asec( ( randu()*1000.0 ) + 1.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/benchmark/benchmark.native.js]: I will recommend to introduce new variable x and then do 
```javascript
x = ( randu()*1000.0 ) + 1.0;
y = asec( x );
```

File: lib/node_modules/@stdlib/math/base/special/asec/benchmark/c/Makefile [hunk 4]
```diff
@@ -0,0 +1,126 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2020 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/benchmark/c/Makefile]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asec/benchmark/c/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,135 @@
+/**
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/benchmark/c/benchmark.c]: we don't need this file, as `math.h` do not have `asec` registered.

File: lib/node_modules/@stdlib/math/base/special/asec/include/stdlib/math/base/special/asec.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/include/stdlib/math/base/special/asec.h]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asec/lib/native.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/lib/native.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asec/lib/native.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var acos = require( '@stdlib/math/base/special/acos' );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/lib/native.js]:
```suggestion
var addon = require( './../src/addon.node' );
```

File: lib/node_modules/@stdlib/math/base/special/asec/lib/native.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+* @example
+* var v = asec( 2.0 );
+* // returns ~1.0472
+*
+* @example
+* var v = asec( NaN );
+* // returns NaN
+*/
+function asec( x ) {
+	return acos( 1.0 / x );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/lib/native.js]:
```suggestion
	return addon( x );
```

File: lib/node_modules/@stdlib/math/base/special/asec/src/addon.c [hunk 10]
```diff
@@ -0,0 +1,23 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/src/addon.c]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 11]
```diff
@@ -0,0 +1,276 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.true( typeof asec, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[-3.0,-1.0]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[-3.0,-1.0]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[1.0,3.0]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[1.0,3.0]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[3.0,100.0]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[3.0,100.0]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[-100.0,-3.0]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[-100.0,-3.0]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 16]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[100.0,1000.0]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[100.0,1000.0]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 17]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[-1000.0,-100.0]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[-1000.0,-100.0]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 18]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[-1e200,-1e208]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[-1e200,-1e208]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 19]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the inverse (arc) secant on the interval `[1e300,1e308]`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function computes the inverse (arc) secant on the interval `[1e300,1e308]`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 20]
```diff
@@ -0,0 +1,276 @@
...
+		} else {
+			delta = abs( y - expected[i] );
+			tol = 1.0 * EPS * abs( expected[i] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function returns `NaN` if provided `NaN`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 21]
```diff
@@ -0,0 +1,276 @@
...
+	var v = asec( NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `+0`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function returns `NaN` if provided `+0`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 22]
```diff
@@ -0,0 +1,276 @@
...
+	var v = asec( 0.0 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `-0`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function returns `NaN` if provided `-0`', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js [hunk 23]
```diff
@@ -0,0 +1,276 @@
...
+	var v = asec( -0.0 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` on the interval `(-1, 1)`', function test( t ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asec/test/test.native.js]:
```suggestion
tape( 'the function returns `NaN` on the interval `(-1, 1)`', opts, function test( t ) {
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Requested changes, once addressed this can be ready.

**Comment by Userf1995** [comment]
@Pranavchiku I have addressed all the issues you mentioned in your review. Please take a look at the changes

**Comment by Userf1995** [comment]
> Requested changes, once addressed this can be ready.

@Pranavchiku I have addressed all the issues you mentioned in your review.please check.

**Review by Planeshifter — APPROVED** [review]
Looks good; let's get this in! Thank you!

**Pull request merged**